ir/memory.cpp

```diff
@@ -195,19 +211,17 @@ expr Pointer::get_offset() const {
   return p.extract(m.bitsOffset() - 1, 0).sextOrTrunc(m.bits_size_t);
 }
 
-expr Pointer::get_value(const char *name, const FunctionExpr &fn,
-                        const expr &ret_type) const {
-  auto bid = get_short_bid();
-  return expr::mkIf(is_local(), fn(bid), expr::mkUF(name, { bid }, ret_type));
-}
```

Comment:
I made Pointer::get_value a template function so that the type of `fn` is parameteric.